x/amm/types/create_pool.go

```diff
@@ -0,0 +1,15 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (msg *MsgCreatePool) CreatePool(ctx sdk.Context, poolID uint64) (*Pool, error) {
+	poolParams := PoolParams{
+		SwapFee: msg.SwapFee,
+		ExitFee: msg.ExitFee,
+	}
+	poolAssets := make([]PoolAsset, len(msg.InitialDeposit))
```

Comment:
I think we should initialize the value of poolAssets. Its simply create n records of empty poolAssets. But later on, in setInitialAssets function, asset.Token is being referenced.